esmvaltool/diag_scripts/weighting/calibrate_sigmas.py

```diff
@@ -0,0 +1,282 @@
+        raise IOError(errmsg)
+    if (settings['performance_sigma_range'][0] >=
+        settings['performance_sigma_range'][1]):
+        errmsg = ' '.join([
+            'performance_sigma_range needs to have the form (min, max) with',
+            'min < max'])
+        raise IOError(errmsg)
+
+    models, _ = read_metadata(cfg)
+
+    performances = {}
+    for variable_group in performance_contributions:
+
+        logger.info('Reading model data for %s', variable_group)
+        datasets_model = models[variable_group]
+        model_data, _ = read_model_data(datasets_model)
+
+        logger.info('Calculating performance for %s', variable_group)
+        performance = calculate_model_distances(model_data)
+        logger.debug(performance.values)
+        performances[variable_group] = performance
+
+    performance = xr.Dataset(performances)
+    overall_performance = compute_overall_mean(
+        performance, performance_contributions)  # (N, N) matrix
+
+    target = models[settings['variable_group_target']]
+    target_data, target_data_files = read_model_data(target)
+    target_data = area_weighted_mean(target_data)
+    # TODO: log_provenance ?
+
+    if cfg['combine_ensemble_members']:
+        # TODO: this combines both dimensions -> I think about if we want this!
+        # other option: only average the model dimension (equivalent to the actual performance weighting)
+        # then do something else with the perfect model dimension (but this will probably make things more
+        # complicated.
+        # - We could run the 80/80 test with all members and then average the in/out count
+        # for the models -> but what about the other metrics?
+        # - We could only use the first member
+        # - We could do some random sampling/ bootstrapping
+
+        overall_performance, groups_performance = combine_ensemble_members(
+            overall_performance)
+        target_data, _ = combine_ensemble_members(target_data)
+
+    # NOTE: what do we do here if 'combine_ensemble_members' is False?
+    overall_performance = mask_dependent_models(
+        overall_performance, overall_independence,
+        settings['most_independent_subset'])
+
+    # TODO: a often used output is the inside_ration versus sigma plot this can
+    # not be provided from this (I think). Use a flag to chose between this and speed?
+    res = minimize(
+        evaluate_future,
+        x0=overall_performance.mean(),
+        bounds=[settings['performance_sigma_range']],
+        args=(overall_performance, target_data, overall_independence,
+              independence_sigma, settings),
+    )
+
+    if res.fun > 999:
+        logmsg = ' '.join([
+            'No confident sigma could be found! Using largest possible value.',
+            'Bad choice of predictors or target or too small sigma range?'
+        ])
+        logger.warning(logmsg)
+
+    return res.x
+
+
+def calibrate_sigmas(independence_sigma: Union[float, str, None],
```

Comment:
I think we need to think a bit more about the 'user interface' in the recipe. Having three potential different options for the input here is a bit too much maybe? Perhaps another input parser could make sense?